src/lib/storage/value_segment.cpp

```diff
@@ -147,26 +116,24 @@ ChunkOffset ValueSegment<T>::size() const {
 
 template <typename T>
 std::shared_ptr<BaseSegment> ValueSegment<T>::copy_using_allocator(const PolymorphicAllocator<size_t>& alloc) const {
-  pmr_concurrent_vector<T> new_values(_values, alloc);  // NOLINT(cppcoreguidelines-slicing)
-                                                        // (clang-tidy reports slicing that comes from tbb)
+  pmr_vector<T> new_values(_values, alloc);  // NOLINT(cppcoreguidelines-slicing)
   if (is_nullable()) {
-    pmr_concurrent_vector<bool> new_null_values(*_null_values, alloc);  // NOLINT(cppcoreguidelines-slicing) (see above)
-    return std::allocate_shared<ValueSegment<T>>(alloc, std::move(new_values), std::move(new_null_values));
+    pmr_vector<bool> new_null_values(*_null_values, alloc);  // NOLINT(cppcoreguidelines-slicing) (see above)
+    return std::make_shared<ValueSegment<T>>(std::move(new_values), std::move(new_null_values));
   } else {
-    return std::allocate_shared<ValueSegment<T>>(alloc, std::move(new_values));
+    return std::make_shared<ValueSegment<T>>(std::move(new_values));
   }
 }
 
 template <typename T>
 size_t ValueSegment<T>::estimate_memory_usage() const {
-  size_t bool_size = 0u;
+  size_t null_values_size = 0u;
   if (_null_values) {
-    bool_size = _null_values->size() * sizeof(bool);
-    // Integer ceiling, since sizeof(bool) equals 1, but boolean vectors are optimized.
-    bool_size = _null_values->size() % CHAR_BIT ? bool_size / CHAR_BIT + 1 : bool_size / CHAR_BIT;
+    null_values_size = _null_values->capacity() % CHAR_BIT ? _null_values->capacity() / CHAR_BIT + 1
```

Comment:
This is duplicated in many places now, isn't it?

Reply:
Yes, but I can't think of an obvious place where this should go. Adding another header makes it harder to understand imho

Reply:
Hm, I see. null_value.hpp?

Reply:
This has only little to do with our representation of SQL's NULL values. As the STL implementation and, thus, this calculation will likely never change, I'd leave it the way it is with the additional benefit of the entire thing being self-contained.

Reply:
Alright